Framework/API/inc/MantidAPI/NexusFileLoader.h

```diff
@@ -20,14 +20,23 @@ class MANTID_API_DLL NexusFileLoader
   virtual void execLoader() = 0; // what would normally be called exec
   // the name of the property that the NexusHDF5Descriptor should be created
   // against
-  virtual std::string getFilenamePropertyName() const = 0;
+  virtual std::string getFilenamePropertyName() const;
   boost::shared_ptr<Algorithm> createChildAlgorithm(
       const std::string &name, const double startProgress = -1.,
       const double endProgress = -1., const bool enableLogging = true,
       const int &version = -1) override;
   virtual void
   setFileInfo(std::shared_ptr<Mantid::Kernel::NexusHDF5Descriptor> fileInfo);
 
+  /// Required to pass m_fileInfo to static functions
+  /// Keeping it shared_ptr to match setFileInfo signature (although passing
+  /// ownership is not the main goal).
+  virtual const std::shared_ptr<Mantid::Kernel::NexusHDF5Descriptor>
```

Comment:
Would we be better off making `m_fileInfo` a protected member variable so that derived classes can access it?

Reply:
yes, the original intention of the getter is to limit the scope for ownership for the usage in this PR. I'd accommodate if there is a preference for getter/setter or protected access, I just added a getter since the setter exists. Let me know.

Reply:
@martyngigg @peterfpeterson What do you recommend here? I'm leaning towards making `m_fileInfo` a protected member variable to reduce the number of function signatures we need to add another variable to, but a getter/setter does work and may have other advantages.

Reply:
Yet another option is to not allow access and just expose the methods people need ;)

I prefer protected and drop the get/set, but it is not a strong preference.

Reply:
we can always address this in a future PR

Reply:
Would we be better having `virtual const Mantid::Kernel::NexusHDF5Descriptor & getFileInfo() const noexcept;`? I'm not sure it's necessary to expose the users of `getFileInfo` to how its memory is managed. Most should not need `set` and having `get` not include the `shared_ptr` gives more freedom to change how that is managed?

Reply:
I'd be happy to make this a separate issue (seems to need more discussion) and carry on before this PR requires rebasing, rebuilding, and testing.